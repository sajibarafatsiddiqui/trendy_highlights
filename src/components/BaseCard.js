import React from 'react';
import { PropTypes } from 'prop-types';

const BaseCard = ({ baseCurrency }) => (
  <div>{ baseCurrency }</div>
);

BaseCard.propTypes = { baseCurrency: PropTypes.string.isRequired };

export default BaseCard;
