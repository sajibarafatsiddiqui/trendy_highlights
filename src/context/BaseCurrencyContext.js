import React from 'react';

const BaseCurrencyContext = React.createContext();
export default BaseCurrencyContext;
