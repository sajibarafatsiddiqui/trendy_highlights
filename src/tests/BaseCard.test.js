import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from 'redux/store';
import BaseCurrencyContext from 'context/BaseCurrencyContext';
import { BrowserRouter as Router } from 'react-router-dom';
import BaseCard from 'components/BaseCard';

describe('Tests for <BaseCard /> component', () => {
  const base = (
    <BaseCard
      baseCurrency="USD"
    />
  );
  const baseCurrency = 'USD';

  const wrap = (component) => (
    <Provider store={store}>
      <BaseCurrencyContext.Provider value={{ baseCurrency }}>
        <Router>
          {component}
        </Router>
      </BaseCurrencyContext.Provider>
    </Provider>
  );
  it('Should render the content correctly', () => {
    render(wrap(base));
    expect(screen.getByText('USD')).not.toBeNull();
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(wrap(base)).toJSON()).toMatchSnapshot();
  });
});
