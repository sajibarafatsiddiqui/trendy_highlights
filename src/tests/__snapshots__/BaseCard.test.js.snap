// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <BaseCard /> component Should match the snapshot 1`] = `
<div
  className="col"
>
  <div
    className="USD BaseCard card"
  >
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      />
      <p
        className="card-text"
      />
      <div
        className="card-footer"
      >
        <h1>
          USD
        </h1>
      </div>
    </div>
  </div>
</div>
`;
